extras/indexing/src/main/java/mvm/rya/indexing/FilterFunctionOptimizer.java

```diff
@@ -99,29 +99,19 @@ public void setConf(final Configuration conf) {
         this.conf = conf;
         //reset the init.
         init = false;
-        try {
             init();
-        } catch (final NumberFormatException | UnknownHostException e) {
-            LOG.error("Unable to update to use new config, falling back to the old config.", e);
```

Comment:
Same reason.  But in addition, this is just a bug and a fib.  It ignores what error happened and any diagnostic information.  It does not restore the old config if there was one. It sets init = false; so that it does not attempt to re-initialize.  The next thing that happens is a failure in distant code that assumes that the database is correctly connected.